### PR TITLE
Sign the chart [ONPREM-440]

### PR DESCRIPTION
**Pull request by christian-stephen**
:gear: **Issue**

*Ticket/s*: https://circleci.atlassian.net/browse/ONPREM-440

:gear: **Change**

Start signing the Helm chart to help ensure provenance: https://helm.sh/docs/topics/provenance/

Change Type: 

AC:

:white_check_mark: **Fix**

:question: **Tests**

- [x] Tests for new feature and update for changes
- [x] Linting passes and/or formatting matches the standard of the repo

🗒️ **Documentation**

- [ ] Updated related documentation (if applicable).
- [x] Updated Change log (if exists)

### CODE REVIEW

File: .circleci/config.yml
Comment (christian-stephen) [inline .circleci/config.yml:1]: Testing: https://app.circleci.com/pipelines/github/CircleCI-Public/container-runner-helm-chart/407/workflows/3dd7d6a6-10bf-44eb-b27b-0b979d0efb8e/jobs/1192?invite=true#step-104-135_27

File: .circleci/config.yml [hunk 1]
```diff
@@ -97,7 +98,26 @@ jobs:
...
+      - run:
+          name: "Install signing keys"
+          command: |
+            exec 2>/dev/null
+
+            echo "Importing signing keys"
+            echo -n "${SIGNING_KEY_ENCODED}" | base64 --decode >signing_key_decoded.key
+            gpg --batch --yes --passphrase "${SIGNING_KEY_PASSPHRASE}" --import signing_key_decoded.key
+            rm signing_key_decoded.key
+            curl https://keys.openpgp.org/vks/v1/by-fingerprint/"${GPG_ID}" >pub-key.asc
```
Comment (ryanwohara) [inline .circleci/config.yml:110]: Does this work as expected? I thought we'd need something like

```
curl -q URL -o pub-key.asc
```

File: do [hunk 2]
```diff
@@ -27,10 +27,26 @@ check-version-bump() {
 help_package="Package the Helm chart"
 package() {
     mkdir -p target
-    cd target
+    cd target || return
```
Comment (ryanwohara) [inline do:30]: Is this `|| return` necessary? I feel like it's going to skip the packaging if we run into an error? But why would we hit an error if we `mkdir -p target` beforehand?

🤔

File: .circleci/config.yml (hunk 1 above)
Comment (christian-stephen) [reply .circleci/config.yml:110]: Yes: https://github.com/circleci/circleci-runner/blob/1e28eb258b3ba2020f7ca992049a6ec86d68d07f/.circleci/config.yml#L971

https://app.circleci.com/pipelines/github/CircleCI-Public/container-runner-helm-chart/407/workflows/3dd7d6a6-10bf-44eb-b27b-0b979d0efb8e/jobs/1192?invite=true#step-104-285_92

File: do (hunk 2 above)
Comment (christian-stephen) [reply do]: This was a left over from some testing I was doing. I've removed it along with addressing a shellcheck warning.

**Review by ryanwohara — APPROVED** [review]

**Pull request merged**